lesson02/main.tf

```diff
@@ -50,14 +48,20 @@ resource "aws_autoscaling_group" "asg" {
   health_check_type         = "${var.health_check_type}"
   health_check_grace_period = "${var.health_check_grace_period}"
   default_cooldown          = "${var.default_cooldown}"
+  min_size                  = "${var.min_size}"
+  max_size                  = "${var.max_size}"
+  wait_for_elb_capacity     = "${var.min_size}"
+  desired_capacity          = "${var.desired_capacity}"
+  vpc_zone_identifier       = ["${data.aws_subnet_ids.vpc_subnets.ids}"]
 
-  min_size              = "${var.min_size}"
-  max_size              = "${var.max_size}"
-  wait_for_elb_capacity = "${var.min_size}"
-
-  desired_capacity = "${var.desired_capacity}"
+  //TAGS propagated to each EC2 instance
+  tags = "${list(
+    map("key", "Name",          "value", "${var.metadata["appname"]}-${var.env}-ec2-${var.metadata["appversion"]}",         "propagate_at_launch", true)
```

Comment:
is this ok? `map("key", "Name",          "value",` with a lot of spaces?

Reply:
let me move them

Reply:
@chitopunk